validation/src/static/scripts/index.ts

```diff
@@ -1 +1,178 @@
+};
+
+/**
+ * Sends call to run a snippet comparison on provided input.
+ * Displays results of that comparison.
+ */
+const getSnippetComparison = async () => {
+  const resultsTable = getElem('resultsTable');
+  const resultsContainer = getElem('snippetResults');
+  if (!resultsTable || !resultsContainer || !editor) return;
+
+  resultsTable.classList.add('hidden');
+  resultsContainer.innerHTML = '';
+
+  const rawResponse = await fetch(
+    'http://localhost:3000/api/runSnippetComparison',
+    {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({snippet: editor.getValue()}),
+    }
+  );
+
+  // Display results in table if response is OK
+  if (rawResponse.status === 200) {
+    const comparisonResults = await rawResponse.json();
+    const accnames = comparisonResults[0];
+
+    getElem('chromeAccName')!.innerText = '"' + accnames.chrome + '"';
+    getElem('axeAccName')!.innerText = '"' + accnames.axe + '"';
+    getElem('aomAccName')!.innerText = '"' + accnames.aom + '"';
+    getElem('bgAccName')!.innerText = '"' + accnames.bg + '"';
+    getElem('ourAccName')!.innerText = '"' + accnames.accname + '"';
+
+    resultsTable.classList.remove('hidden');
+
+    if (comparisonResults[1]) {
+      const caseId = comparisonResults[1];
+
+      resultsContainer.innerHTML += `
+        <div class="bluetext comparisonResultText">A test-case was generated as a result of this comparison - <a class="linkBtn" href="/case/${caseId}">View Test-case ${caseId}</a></div>
+      `;
+    }
+  } else if (rawResponse.status === 400) {
+    const error = await rawResponse.json();
+    resultsContainer.innerHTML = `<div class="redtext comparisonResultText">Error: ${error.message}</div>`;
+  }
+};
+
+/**
+ * Sends call to run a URL comparison on provided input URL.
+ * Displays results of that comparison.
+ */
+const getURLComparison = async () => {
+  const urlInput = getElem('urlInput') as HTMLInputElement;
+  const urlResults = getElem('urlResults');
+  if (!urlInput || !urlResults) return;
+  urlResults.innerHTML = '';
+
+  // Adds an animated CSS Loading Spinner
+  const spinner = document.createElement('div');
+  spinner.classList.add('loader');
+  urlResults.append(spinner);
+
+  const url = urlInput.value;
+  const rawResponse = await fetch(
+    'http://localhost:3000/api/runURLComparison',
+    {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({url: url}),
+    }
+  );
+
+  // Remove spinner once we get a response
+  spinner.remove();
+  if (rawResponse.status === 200) {
+    const summaryId = await rawResponse.json();
+    urlResults.innerHTML = `<div class="bluetext comparisonResultText">View the comparison results for ${url} - <a class="linkBtn" href="/summary/${summaryId}">Summary ${summaryId}</a></div>`;
```

Comment:
It would be useful if these (+snippets) were added to the table at this point. If I evaluate one URL then another, I have no way to view the results of the first wihtout refreshing